engine/session_live.js

```diff
@@ -994,53 +959,61 @@ class SessionLive {
   }
 
   _addLiveSegmentsToQueue(startIdx, playlistItems, baseUrl, liveTargetBandwidth) {
-    let incrementDiscSeqCount = false;
+    let amountRemovedDiscSeqs = 0;
     for (let i = startIdx; i < playlistItems.length; i++) {
       debug(`[${this.sessionId}]: Adding Live Segment(s) to Queue (for bw=${liveTargetBandwidth})`);
       let seg = {};
       let playlistItem = playlistItems[i];
       let segmentUri;
+      let cueData = null;
+      let daterangeData = null;
       let attributes = playlistItem["attributes"].attributes;
-
       if (playlistItem.properties.discontinuity) {
         this.liveSegQueue[liveTargetBandwidth].push({ discontinuity: true });
         this.liveSegsForFollowers[liveTargetBandwidth].push({ discontinuity: true });
       }
       if ("cuein" in attributes) {
-        this.liveSegQueue[liveTargetBandwidth].push({ cue: { in: true } });
-        this.liveSegsForFollowers[liveTargetBandwidth].push({ cue: { in: true } });
```

Comment:
keeping segment attributes as individual seg-Items would also work, however before, we only compensated the Segment Queue Shifting process for discontinuity tags and cue tags.  Meaning that it would shift inaccurately if a daterange attribute was found. The shifting process is simpler if we combine them like in hls-vodtolive